### PR TITLE
[+] Card Implementation: Rush Clovis & Clear the Field Implementation

### PR DESCRIPTION
**Pull request by HugoHMZ**
image
image

### CODE REVIEW

File: server/game/cards/03_TWI/units/ClearTheField.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+            id: '9620454519',
+            internalName: 'clear-the-field'
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.setEventAbility({
+            title: 'Choose a non-leader that costs 3 or less. Return it and each enemy non-leader unit with the same name as it to their owner\'s hand.',
+            targetResolver: {
+                cardTypeFilter: WildcardCardType.NonLeaderUnit,
```
Comment (AMMayberry1) [inline server/game/cards/03_TWI/units/ClearTheField.ts:17]: nit: we can remove this and just do `card.isNonLeaderUnit()` in the condition

File: server/game/cards/03_TWI/units/ClearTheField.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+                zoneFilter: WildcardZoneName.AnyArena,
+                controller: WildcardRelativePlayer.Any,
```
Comment (AMMayberry1) [inline server/game/cards/03_TWI/units/ClearTheField.ts:19]: nit: these are the default values for zoneFilter and controller so we can remove them

File: test/server/cards/03_TWI/units/ClearTheField.spec.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+            context.player1.clickCard(specforceSoldiersP2[0]);
+            expect(specforceSoldiersP2[0]).toBeInZone('hand', context.player2);
+            expect(specforceSoldiersP2[1]).toBeInZone('hand', context.player2);
+
+            // Check ally card that matches the target name doesn't go back to hand
+            expect(specforceSoldierP1).toBeInZone('groundArena', context.player1);
+
+            // Check card that doesn't match the selected card's name stays in play
+            expect(context.duchesssChampion).toBeInZone('groundArena');
+            expect(context.yoda).toBeInZone('groundArena');
```
Comment (AMMayberry1) [inline test/server/cards/03_TWI/units/ClearTheField.spec.ts:31]: Good test! Few more test cases it would be good to have:
- Test with two different versions of a unique unit to confirm that cards will match even if the subtitles are different
- Try with a leader whose name matches to confirm that it doesn't try to return it to hand and break the game
- Apply it to some token units to confirm they are all removed from the game

File: server/game/cards/03_TWI/units/RushClovis.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    protected override getImplementationId () {
+        return {
+            id: '9964112400',
+            internalName: 'rush-clovis#banking-clan-scion'
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.addOnAttackAbility({
+            title: 'If the defending player controls no ready resources, create a Battle Droid token.',
```
Comment (AMMayberry1) [inline server/game/cards/03_TWI/units/RushClovis.ts:14]: nit: we don't include the "if" part in the title since this prompt will only appear if the condition is already met

File: server/game/cards/03_TWI/units/RushClovis.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+            immediateEffect: AbilityHelper.immediateEffects.conditional({ condition: (context) => context.source.controller.opponent.resources.filter((resource) => !resource.exhausted).length === 0,
```
Comment (AMMayberry1) [inline server/game/cards/03_TWI/units/RushClovis.ts:15]: nit: can do `context.source.controller.opponent.readyResourceCount`

File: test/server/cards/03_TWI/units/RushClovis.spec.ts [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+                player2: {
+                    groundArena: ['rush-clovis#banking-clan-scion'],
+                    resources: 1
+                }
+            });
+            const { context } = contextRef;
+            const rushClovisP1 = context.player1.findCardByName('rush-clovis#banking-clan-scion');
+            const rushClovisP2 = context.player2.findCardByName('rush-clovis#banking-clan-scion');
+
+            // Check player 1 doesn't get a Battle Droid token since player 2 has ready resources.
```
Comment (AMMayberry1) [inline test/server/cards/03_TWI/units/RushClovis.spec.ts:19]: Zero resources is kind of a weird edge case, could we instead test with a nonzero number of exhausted resources? Can use `context.player1.exhaustResources(5);` in the test to accomplish that
